### PR TITLE
fix: enable pod logging when waiting times out

### PR DESCRIPTION
**Pull request by ChrisPRobinson**
## What this PR does / why we need it
Enables more descriptive logging when waiting for pods to complete fails

## Jira ID

[DTSS-1735]

## Notes for your reviewers

[DTSS-1735]: https://outreach-io.atlassian.net/browse/DTSS-1735?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ChrisPRobinson** [comment]
Need to do a final e2e test of this on Monday

File: pkg/devenvutil/devenv.go [hunk 1]
```diff
@@ -209,34 +210,97 @@ func FindUnreadyPods(ctx context.Context, k kubernetes.Interface) ([]string, err
...
+
 	return ctx.Err()
 }
+
+// pod is a wrapper around []*corev1.Pod so that it can marshal to
+// a structured log
+type pods []*corev1.Pod
+
+// MarshalLog implements the log.Marshaler interface for []*corev1.Pod
+func (p *pods) MarshalLog(addField func(key string, v interface{})) {
```
Comment (jaredallard) [inline pkg/devenvutil/devenv.go:260]: Does this actually work? We don't use the gobox logger, it'd be cool if it did though

Comment (ChrisPRobinson) [reply pkg/devenvutil/devenv.go:260]: ugh I made terrible assumptions. the marshaling works but my integration of course does not...

Comment (ChrisPRobinson) [reply pkg/devenvutil/devenv.go]: @jaredallard fixed.

File: pkg/devenvutil/devenv.go [hunk 2]
```diff
@@ -209,34 +210,90 @@ func FindUnreadyPods(ctx context.Context, k kubernetes.Interface) ([]string, err
...
-		log.WithError(err).WithField("pods", unreadyPods).
+		log.WithError(err).WithField("pods", unreadyPodNames).
 			Info("Waiting for pods to be ready")
 
 		async.Sleep(ctx, 30*time.Second)
 	}
 
+	if ctx.Err() != nil {
+		// Write out a bit more detailed info on the prior to exit
+		log.WithError(err).WithField("pods", PodsStateInfo(unreadyPods)).
```
Comment (ChrisPRobinson) [inline pkg/devenvutil/devenv.go:247]: It might be better to change the field name and message, given this will be written out after a log.WithError.

**Review by jaredallard — APPROVED** [review]

**Pull request merged**